src/transformers/modeling_utils.py

```diff
@@ -1777,10 +1778,15 @@ def save_pretrained(
                 # Removing the keys which are declared as known duplicates on
                 # load. This allows to make sure the name which is kept is consistent.
                 if self._keys_to_ignore_on_load_missing is not None:
+                    del_names = set()
                     for name in names:
                         matches_pattern = any(re.search(pat, name) for pat in self._keys_to_ignore_on_load_missing)
                         if matches_pattern and name in state_dict:
-                            del state_dict[name]
+                            del_names.add(name)
+                    # This makes sure even if the pattern covers all names
+                    # that we keep at least 1 copy of the name.
+                    for name in sorted(del_names)[: len(names) - 1]:
+                        del state_dict[name]
```

Comment:
This is the fix for Deta.

Deta `_keys_to_ignore_on_load` regexp are a bit too generous and cover, ALL duplicates for some layers.
This code ensures that we keep at least 1 key in the dict in that case.

Reply:
I am feeling this part might need a bit more verification: for other models (i.e. not `deta`), is it possible that we leave more than 1 duplicates with this new change?

Reply:
In terms of logic, I think we should keep the first one and not the last? Usually tensor sharing is written as `tensor_2 = tensor_1` not the opposite.

Reply:
(I guess you somehow saw my deleted message?) I think the current change is fine and so I deleted my previous question